docs/developer/mesh.md

```diff
@@ -0,0 +1,270 @@
+        |       ├── vecs (G)
+        |          ├── coordinates (G) Attribute (1) blockSize
+        |             |
+        |             ├── coordinates (D, FLOAT, size = 3 * nvertices) Note: Each vertex has 3 coordinates in x,y,z
+        |
+        ├── labels
+        |   ├── Cell Sets (G)
+        |   |  ├── 1 (G)
+        |   |     ├── indicies (D, INTEGER, size = ncells) [0:ncells-1]
+        |   |
+        |   ├── Face Sets (G, OPTIONAL)
+        |   |  ├── 1 (G)
+        |   |  |  ├── indicies (D, INTEGER) (e.g. [31 37] for right boundary)
+        |   |  |
+        |   |  ├── 2 (G)
+        |   |     ├── indicies (D, INTEGER) (e.g. [28 36] for left boundary)
+        |   |  |
+        |   |  ├── 3 (G)
+        |   |     ├── indicies (D, INTEGER) (e.g. [35 38 49] for top boundary)
+        |   |  |
+        |   |  ├── 4 (G)
+        |   |     ├── indicies (D, INTEGER) (e.g. [26 30 40] for bottom boundary)
+        |   |  |
+        |   |  ├── 5 (G)
+        |   |     ├── indicies (D, INTEGER) (e.g. [32 42 45 46 47] for river)
+        |   |
+        |   ├── boundary_edges (G)
+        |   |  ├── 1 (G)
+        |   |     ├── indicies (D, INTEGER, size = nedges_bnd)
+        |   |
+        |   ├── celltype (G)
+        |      ├── 0 (G)
+        |      |  ├── indicies (D, INTEGER, size = num_verrtices) [ncells:ncells+nvertices-1]
+        |      |
+        |      ├── 1 (G)
+        |      |  ├── indicies (D, INTEGER, size = num_edges)     [ncells+nvertices:ncells+nvertices+nedges-1]
+        |      |
+        |      ├── 3 (G, OPTIONAL, TRIANGULAR)
+        |      |  ├── indicies (D, INTEGER, size = num_tri)       [nquads:nquads+ntri-1]
+        |      |
+        |      ├── 4 (G, OPTIONAL, QUADRILATERALS)
+        |         ├── indicies (D, INTEGER, size = num_quads)     [0:nquads]
+        |
+        ├── topology (G) Attributes: (1) cell_dim (INTEGER) (2) depth (INTEGER)
+            |
+            ├── permutation (D, INTEGER, size = 3)
+            |
+            ├── strata (G)
+               ├── 0 (G)
+               |   ├── cone_sizes  (D, INTEGER, size = num_vertices)
+               |   ├── cones       (D, INTEGER)
+               |   ├── orientation (D, INTEGER)
+               |
+               ├── 1 (G)
+               |   ├── cone_sizes  (D, INTEGER, size = num_edges)
+               |   ├── cones       (D, INTEGER, size = 2 * num_edges)
+               |   ├── orientation (D, INTEGER, size = 2 * num_edges)
+               |
+               ├── 2 (G)
+                   ├── cone_sizes  (D, INTEGER, size = ncells)
+                   ├── cones       (D, INTEGER, size = 4 * nquads + 3 * ntri)
+                   ├── orientation (D, INTEGER, size = 4 * nquads + 3 * ntri)
+
+```
+
+## Domain Decomposition
+
+Below is the decomposition of the example mesh with mixed element types across two MPI ranks.
+
+![image](figures/quad_tri_mesh_dmplex_numbering_N2.png)
+
+The SWE physics in RDycore is currently implmented in two versions:
```

Comment:
I'm not sure we need to mention SWE specifically here--we're likely to duplicate this structure in all of our physics. We could skip the PETSc implementation going forward, but personally I think we'll want it as a simpler reference implementation.

Reply:
"_SWE_" has been removed and I agree that we will continue to keep the PETSc implementation as it is simpler to debug.